garbage-collection-overview.md

```diff
@@ -13,31 +13,34 @@ TiDB uses MVCC to control transaction concurrency. When you update the data, the
+The TiDB transaction model is implemented based on [Google's Percolator](https://ai.google/research/pubs/pub36726). It's mainly a two-phase commit protocol with some practical optimizations. When the first phase is finished, all the related keys are locked. Among these locks, one is the primary lock and the others are secondary locks which contain a pointer to the primary lock; in the second phase, the key with the primary lock gets a write record and its lock is removed. The write record indicates the write or delete operation in the history or the transactional rollback record of this key. The type of write record that replaces the primary lock indicates whether the corresponding transaction is committed successfully. Then all the secondary locks are replaced successively. If, for some reasons such as failure, the threads fail to replace the secondary locks and these locks are retained, you can still find the primary key based on the information contained in the secondary locks and determines whether the entire transaction is committed based on whether the primary key is committed. However, if the primary key information is cleared by GC and this transaction has uncommitted secondary locks, you will never learn whether these lock should be committed. As a result, data integrity cannot be guaranteed.
 
-The Resolve Locks step rolls back or commits the locks before the safe point, depending on whether their primary key has been committed or not. If the primary key is also retained, the transaction times out and is rolled back.
-This step is required. Once GC has cleared the write record of the primary lock, you can never know whether this transaction is successful or not. Also, if the transaction contains retained secondary keys, it's important to know whether it should be rolled back or committed. As a result, data consistency cannot be guaranteed.
+The Resolve Locks step clears the locks before the safe point. This means that if the corresponding primary key of a lock is committed, this lock should be committed; otherwise, it should be rolled back. If the primary key is still locked (not committed or rolled back), this transaction is seen as timing out and rolled back.
```

Comment:
```suggestion
The Resolve Locks step clears the locks before the safe point. This means that if the primary key of a lock is committed, this lock needs to be committed; otherwise, it needs to be rolled back. If the primary key is still locked (not committed or rolled back), this transaction is seen as timing out and rolled back.
```